corehq/form_processor/submission_post.py

```diff
@@ -647,3 +658,18 @@ def notify_submission_error(instance, message, exec_info=None):
     }
     request = get_request()
     notify_exception(request, message, details=details, exec_info=exec_info)
+
+
+@quickcache(['domain', 'build_id'])
+def _get_form_name_info(domain, build_id):
+    """
+    :return: (default_language, {'http://my.xmlns': {'en': 'My Form'}})
+    """
+    try:
+        app_build = get_current_app(domain, build_id)
+    except ResourceNotFound:
```

Comment:
Out of curiosity, what situations would/could cause this exception?

Reply:
Certainly this could happen if `build_id` was None or blank which would be the case for form submitted internally or via integrations. @esoergel what about short circuiting this process so that we don't call this method for forms that don't have a `build_id`?

Reply:
Turns out passing a blank ID actually raises a different exception than a missing ID - added in a short-circuit for that.  @nospame this could also occur if the app build can't be found for whatever reason.  Turns out deleting a build _won't_ actually trigger this, because apps get soft deleted, but in this workflow I want to be extra cautious, so I'd like to leave in the error handling.  This success message is a nice-to-have, but there's a reasonable fallback, and causing form submissions to error would be very bad.  If this were more core to the workflow, I probably wouldn't code so defensively.